### PR TITLE
Generate example sdk using gradle

### LINKED ISSUE #5: Add tooling for invoking code generation
In #4 I started to lean in on gradle for coordinating everything about our code generation.

Some of the discussion on the PR has led me to think that gradle is probably not the right tool for coordinating dynamic methods of generating the code.  Instead, we can use gradle for management of jar dependencies and then either use bash or `npm` for actually calling the codegen commands.

Since we're using npm already for most of the pdc it seems reasonable to just leverage npm with a package.json as our point of entry.

### PR DESCRIPTION
**Pull request by slifty**
This PR sets up a "hello world" generator in the typescript namespace and adds the appropriate gradle configuration for invoking the generator.

It also updates our editorconfig to use tabs, as we're moving other aspects of the project to tabs.

You should be able to test this by following the instructions in the README (ultimately invoking `./gradlew generateSwaggerCode to see the generated content)

Resolves #5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,32 @@
-# sdk
-Instructions and supporting files for SDK generation
+# PDC SDK Generators
+
+This repository houses the infrastructure for generating PDC SDKs.
+
+We're using a project called [swagger-codegen](https://github.com/swagger-api/swagger-codegen) to power our SDK, and this repository provides tooling and templates
+to make sdk generation easier in particular target languages.  The codegen project supports a [many languages](https://github.com/swagger-api/swagger-codegen/tree/master/modules/swagger-codegen/src/main/resources)
```
Comment (reefdog) [inline README.md]:
```suggestion
to make sdk generation easier in particular target languages. The codegen project supports a [many languages](https://github.com/swagger-api/swagger-codegen/tree/master/modules/swagger-codegen/src/main/resources)
```

Comment (reefdog) [reply README.md]: (This is not a real suggestion, so much as a poetic lament.)

File: README.md [hunk 2]
```diff
@@ -1,2 +1,32 @@
...
+
+2. Populated configuration files, for each language, located in the `configuration` directory.
+
+#### Steps:
+
+From the project root you can run the following commands:
+
+```shell
+> cp YOUR_SWAGGER_SPEC_LOCATION ./openapi.json
+> envsubst ./configuration/typescript.json.template>./configuration/typescript.json
```
Comment (bickelj) [inline README.md]: This command appeared to wait for more input on my machine.

Comment (slifty) [inline README.md]: This line should be: `envsubst < ./configuration/typescript.json.template > ./configuration/typescript.json `

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,28 @@
+plugins {
+	id 'base'
+	id 'org.hidetake.swagger.generator' version '2.19.2'
+}
+
+repositories {
+		mavenCentral()
```
Comment (bickelj) [inline build.gradle]: Whitespace here seems off.

**Review by bickelj — APPROVED** [review]
This looks way gradleyer than before! Nice!

Minor suggestions above.

File: README.md (hunk 1 above)
Comment (slifty) [reply README.md]: I do what prettier tells me to do justin, your wish is granted.

**Comment by slifty** [comment]
w00t thank you @bickelj!  Addressed the suggestions, and also reorganized the commits to separate the custom template from the initial setup.  Merging in and going forward to actually write custom templates!

File: README.md (hunk 1 above)
Comment (reefdog) [reply README.md]: 😮 Prettier removed the double-spaces?? I LOVE PRETTIER

Comment (slifty) [reply README.md]: It did in the other repo -- https://github.com/PhilanthropyDataCommons/service/pull/695#discussion_r1443550634

(we aren't running it here, but I try to abide by the demand of automated tools in all aspects of life)

Comment (reefdog) [reply README.md]: I love that Prettier has more credibility than _me_.

**Pull request merged**